gateway/mw_external_oauth.go

```diff
@@ -80,8 +87,93 @@ func (k *ExternalOAuthMiddleware) ProcessRequest(w http.ResponseWriter, r *http.
 
 // jwt makes access token validation without making a network call and validates access token locally.
 // The access token should be JWT type.
-func (k *ExternalOAuthMiddleware) jwt(accessToken string) (bool, string, error) {
-	return false, "", errors.New("jwt validation not implemented yet")
+func (k *ExternalOAuthMiddleware) jwt(accessToken string, jwtValidation apidef.JWTValidation) (bool, string, error) {
+	parser := jwt.NewParser(jwt.WithoutClaimsValidation())
+	// Verify the token
+	token, err := parser.Parse(accessToken, func(token *jwt.Token) (interface{}, error) {
+		// Don't forget to validate the alg is what you expect:
+		if err := assertSigningMethod(k.Spec.JWTSigningMethod, token); err != nil {
+			return nil, err
+		}
+
+		val, err := k.getSecretFromJWKOrConfig(token.Header[KID].(string))
```

Comment:
is it possible that `nil` value is tried to be asserted as `string`?

Reply:
yes, I'll add the typecast here